assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -0,0 +1,215 @@
+		} else {
+			contextSlugsAsArray = [ ...contextSlugsAsArray ];
+		}
+
+		return {
+			payload: { areaSlug, contextSlugs: contextSlugsAsArray },
+			type: ASSIGN_WIDGET_AREA,
+		};
+	},
+
+	/**
+	 * Creates a widget area with a given name and settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}             areaSlug           Widget Area's slug.
+	 * @param  {Object}             settings           Widget Area's settings.
+	 * @param  {number}             settings.priority  Priority for this widget area. Default: 10.
+	 * @param  {string}             settings.title     Title for this widget area.
+	 * @param  {string}             settings.subtitle  Subtitle for this widget area.
+	 * @param  {(string|undefined)} settings.icon      URL to SVG icon for this widget area.
+	 * @param  {string}             settings.style     Widget area style (one of "boxes", "composite").
+	 * @return {Object}                                Redux-style action.
+	 */
+	registerWidgetArea( areaSlug, settings ) {
+		invariant( areaSlug, 'areaSlug is required.' );
+		invariant( settings, 'settings is required.' );
+		invariant( settings.title, 'settings.title is required.' );
+		invariant( settings.subtitle, 'settings.subtitle is required.' );
+		invariant( settings.style, 'settings.style is required.' );
+
+		const settingsToUse = {
+			priority: 10,
+			...settings,
+		};
+
+		return {
+			payload: { areaSlug, settings: settingsToUse },
+			type: REGISTER_WIDGET_AREA,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case ASSIGN_WIDGET_AREA: {
+			const { areaSlug, contextSlugs } = payload;
+
+			const { contexts } = state;
+			contextSlugs.forEach( ( contextSlug ) => {
+				if ( contexts[ contextSlug ] === undefined ) {
+					return;
+				}
+
+				if ( ! contexts[ contextSlug ].includes( areaSlug ) ) {
+					contexts[ contextSlug ].push( areaSlug );
+				}
+			} );
+
+			return {
+				...state,
+				contexts,
+			};
+		}
+
+		case REGISTER_WIDGET_AREA: {
+			const { areaSlug, settings } = payload;
+
+			if ( state.areas[ areaSlug ] !== undefined ) {
+				global.console.warn( `Could not register widget area with slug "${ areaSlug }". Widget Area "${ areaSlug }" is already registered.` );
+
+				return { ...state };
+			}
+
+			return {
+				...state,
+				areas: {
+					...state.areas,
+					[ areaSlug ]: { ...settings, slug: areaSlug },
+				},
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+};
+
+export const selectors = {
+	/**
+	 * Checks if a widget area has been registered.
+	 *
+	 * Returns `true` if the widget area has been registered.
+	 * Returns `false` if the widget area has NOT been registered.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {Object} state    Data store's state.
+	 * @param  {string} areaSlug Widget Area's slug.
+	 * @return {boolean}         `true`/`false` based on whether widget area has been registered.
```

Comment:
No need to indent the description here, this should directly follow the type hint. Aligned indentation should only apply to parameters, returns are separate.

Reply:
I found it a lot easier to read this way, but fair enough! 🙂